@xen-orchestra/self-signed/readCert.js

```diff
@@ -0,0 +1,81 @@
+'use strict'
+
+const { dirname } = require('node:path')
+const { mkdir, readFile, writeFile, unlink } = require('node:fs/promises')
+const { X509Certificate } = require('node:crypto')
+
+const { genSelfSignedCert } = require('./index.js')
+
+const identity = value => value
+
+const noop = Function.prototype
+
+async function outputFile(path, content, mode) {
+  for (let attempt = 0; attempt < 5; ++attempt) {
+    try {
+      return await writeFile(path, content, { mode })
+    } catch (error) {
+      const { code } = error
+      if (code === 'ENOENT') {
+        await mkdir(dirname(path), { mode, recursive: true })
+      } else if (code === 'EACCES') {
+        await unlink(path)
+      } else {
+        throw error
+      }
+    }
+  }
+}
+
+exports.readCert = async function readCert(
+  certPath,
+  keyPath,
+  {
+    autoCert = false,
+    use = identity,
+    info = noop,
+    mode = 0o400,
+    warn = noop,
+
+    ...opts
+  }
+) {
+  let useError = false
```

Comment:
```suggestion
  let readingDone = false
```
we don't know if there will be an error during use, but we do want to handle differently error thrwon during read/generation and after